notebook/services/contents/tests/test_manager.py

```diff
@@ -129,6 +129,26 @@ def test_bad_symlink(self):
             # broken symlinks should still be shown in the contents manager
             self.assertTrue('bad symlink' in contents)
 
+    @dec.skipif(sys.platform == 'win32' and sys.version_info[0] < 3)
+    def test_recursive_symlink(self):
+        with TemporaryDirectory() as td:
+            cm = FileContentsManager(root_dir=td)
+            path = 'test recursive symlink'
+            _make_dir(cm, path)
+            os_path = cm._get_os_path(path)
+            os.symlink("recursive", os.path.join(os_path, "recursive"))
+            file_model = cm.new_untitled(path=path, ext='.txt')
+
+            model = cm.get(path)
+
+            contents = {
+                content['name']: content for content in model['content']
+            }
+            self.assertIn('untitled.txt', contents)
+            self.assertEqual(contents['untitled.txt'], file_model)
+            # recursive symlinks should not be shown in the contents manager
+            self.assertNotIn('recursive', contents)
```

Comment:
I fixed the spelling, but this test is now failing on Windows:

```
======================================================================
FAIL: test_recursive_symlink (notebook.services.contents.tests.test_manager.TestFileContentsManager)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Miniconda36-x64\lib\site-packages\ipython_genutils\testing\decorators.py", line 186, in skipper_func
    return f(*args, **kwargs)
  File "C:\projects\notebook\notebook\services\contents\tests\test_manager.py", line 150, in test_recursive_symlink
    self.assertNotIn('recursive', contents)
AssertionError: 'recursive' unexpectedly found in {'recursive': {'name': 'recursive', 'path': 'test recursive symlink/recursive', 'last_modified': datetime.datetime(2019, 6, 25, 8, 35, 46, 553212, tzinfo=<notebook._tz.tzUTC object at 0x000000B0AB495D68>), 'created': datetime.datetime(2019, 6, 25, 8, 35, 46, 553212, tzinfo=<notebook._tz.tzUTC object at 0x000000B0AB495D68>), 'content': None, 'format': None, 'mimetype': None, 'size': 0, 'writable': True, 'type': 'file'}, 'untitled.txt': {'name': 'untitled.txt', 'path': 'test recursive symlink/untitled.txt', 'last_modified': datetime.datetime(2019, 6, 25, 8, 35, 46, 553212, tzinfo=<notebook._tz.tzUTC object at 0x000000B0AB495D68>), 'created': datetime.datetime(2019, 6, 25, 8, 35, 46, 553212, tzinfo=<notebook._tz.tzUTC object at 0x000000B0AB495D68>), 'content': None, 'format': None, 'mimetype': 'text/plain', 'size': 0, 'writable': True, 'type': 'file'}}
```

I guess Windows doesn't handle this in the same way Unix does. It's probably fine to just skip the test on Windows completely. I think it's unusual to use symlinks on Windows anyway.